### ISSUE TITLE
[Bug] Can't Use Address Bar while Snackbar is Displayed

### ISSUE THREAD

**Issue by theFireProject** [mozilla-mobile/fenix#9539]
## Steps to reproduce
**Scenarios**
This bugs only occurs for specific snackbars, not for all. From what I've found out, the bug occurs at least for these two:
- "URL copied to clipboard" snackbar
    - go to a website
    - tap and hold the address on the address bar
    - tap 'Copy' to copy the address
- "Deleted «top-level domain»" snackbar
    - go to a website
    - save it as a bookmark
    - edit the saved bookmark
    - delete the bookmark
- try to tap the address to edit it or try to tap another button on the address bar

The bug does not take effect for these snackbars:
- "Bookmark saved! 'EDIT'" snackbar
- "New tab opened 'SWITCH'" snackbar
- "Tab closed 'UNDO'" snackbar
- "Added to top sites!" snackbar

### Expected behavior
After copying the address, the confirmation snackbar is displayed saying "URL copied to clipboard".
It's displayed above the address bar without overlapping it. So the address bar is fully accessible in theory. Tapping the address field is supposed to open the keyboard to edit the URL and tapping other buttons on the address bar should trigger the right action, too.

### Actual behavior
The address bar is not useable while the confirmation snackbar "URL copied to clipboard" is displayed. There is no reaction to tapping anywhere on the address bar.

### Device information

* Android device: Samsung Galaxy S9+ (Android 8.0.0)
* Fenix version: 4.2.0

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-16533)

**Comment by LaurentiuApahideanSV** [comment]
I can confirm that this issue is reproducible using the latest version of Firefox Preview Nightly 3/31 (Build #20910607).

Devices:
OnePlus 6T (Android 9);
Huawei MediaPad M3 lite 10 (Android 7.0)

**Comment by sblatz** [comment]
This is happening in cases where we display the snackbar on `ContentFrameLayout` (which we need to do in order to display it while navigating to another fragment). I'm not sure what options we really have here other than getting rid of those types of snackbars.

**Comment by SL-Lee** [comment]
As of **Nightly 200722 06:08 (Build #22040614)**, I have discovered three more snackbars that not only obstructs the address bar while it is displayed, it also cannot be dismissed by swiping right on it:

* "Tab sent!" snackbar: appears when sending a tab to another device (issue mentioned [here](https://github.com/mozilla-mobile/fenix/issues/11199#issuecomment-648445644)).
* "Private tab closed" snackbar: appears when closing the last private tab. Note that this only applies when you are closing the **last** tab (#11851)
* "Successfully uninstalled {add-on name}" snackbar: appears when you uninstall an add-on. This snackbar also covers/overlays the toolbar, unlike others which appear above the toolbar (#9196),

Please fix these issues. Having to wait for the snackbar to disappear before the toolbar is usable becomes really annoying after a while.

**Comment by opusforlife2** [comment]
> "New tab opened 'SWITCH'" snackbar

I was testing this, and I noticed that this snackbar appears _right on top_ of the address bar as of version Nightly 200723 06:01 (Build #22050611). Is that a regression or was it always the case? If so, it should be changed to be consistent.

**Comment by person808** [comment]
>I was testing this, and I noticed that this snackbar appears right on top of the address bar as of version Nightly 200723 06:01 (Build #22050611). Is that a regression or was it always the case? If so, it should be changed to be consistent.

Swipe-to-switch tabs broke a lot of snackbars and some slipped through the cracks. I'll break this one out into another issue since I think I have a fix.

> "Tab closed"/"Private tab closed" snackbar: appears when closing a normal/private tab by long-pressing on the tab icon and tapping on "Close tab". This snackbar also covers/overlays the toolbar (#12868).

There is a fix for this one. See the issue I edited in

**Comment by Mugurell** [comment]
@ QA I think all of the issues with the snackbar appearing over the toolbar are now fixed.
Are there any others left or can this ticket now be closed?

**Comment by softvision-miralobontiu** [comment]
Verified as fixed on the 2/8 Nightly build with Samsung Galaxy Note 8 (Android 9).

**Comment by SL-Lee** [comment]
The "Tab sent!" snackbar (which appears when you send a tab to a device) still renders the toolbar unusable while it is displayed (I'm using **Nightly 210207 17:02 Build #2015791947**).

**Comment by Mugurell** [comment]
> The "Tab sent!" snackbar (which appears when you send a tab to a device) still renders the toolbar unusable while it is displayed (I'm using **Nightly 210207 17:02 Build #2015791947**).

Can you tell us in which situation?
A video of the issue would help greatly!

**Comment by SL-Lee** [comment]
The "Tab sent!" snackbar appears when you visit a web page, and send that tab to another device. See the GIF attached below -- after sending a tab to another device, repeatedly tapping the address bar while the snackbar is displayed does nothing. The toolbar becomes accessible only after the snackbar disappears.

![](https://user-images.githubusercontent.com/45960387/107260093-bcbc2a00-6a78-11eb-8075-d81dd058f946.gif)

**Comment by leadweedy** [comment]
The issue is present in an additional case of the "Deleted «top-level domain»" snackbar when removing bookmarks.
It's affecting both Nightly (90.0a1 (Build #2015808459)) and regular release (88.1.2 (Build #2015806939)). I can also reproduce the effect with the "Tab sent!" snackbar. None of the other cases mentioned in this thread seem to be affected. 

https://user-images.githubusercontent.com/72583854/117117907-f41f3780-ad44-11eb-83ca-6e944ffda528.mp4

**Comment by SL-Lee** [comment]
I still can reproduce this issue on **Firefox Beta 94.0.0-beta.5 (Build 2015841163)** and **Firefox Nightly 95.0a1 (Build #2015843019)** using the same steps as stated in [my previous comment](https://github.com/mozilla-mobile/fenix/issues/9539#issuecomment-775324677).

**Comment by vitaminj** [comment]
The double tap issue appears to have been fixed in 103 but the main issue still remains

**Comment by csadilek** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1807135

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).
